src/main/java/uk/gov/digital/ho/hocs/audit/export/infoclient/ExportViewConstants.java

```diff
@@ -2,11 +2,23 @@
 
 public interface ExportViewConstants {
 
+        enum DATE_SECTION {DAY, MONTH, YEAR};
+        int YEAR_DIGITS = 4;
+        int MONTH_DIGITS = 2;
+        int DAY_DIGITS = 2;
+
+        // identifies dates that have the format [0]YYYY-[0]MM-[0]DD
+        String MALFORMED_DATE_REGEX = "^0*\\d{4}\\-0*(?:[1-9]|1[0-2])\\-0*(?:[1-9]|[12][0-9]|3[01])$";
```

Comment:
High: This regex can include groups i.e. `^(0*\d{4})\-(0*(?:[1-9]|1[0-2]))\-(0*(?:[1-9]|[12][0-9]|3[01]))$` which would then extract the individual matched parts. It would also mean not having separate regexes (EXTRACT_*) that may differ/contradict the original pattern sought.

Reply:
The regex was originally `^0*\d{4}\-0*(?:[0-9]|1[0-2])\-0*(?:[1-9]|[12][0-9]|3[01])$` and was changed to the above based on an earlier comment.

Reply:
Could this not also work? 
`^0*(\d{4})\-0*([0-9]|1[0-2])\-0*([1-9]|[12][0-9]|3[01])$`

It would match the date and also group and pull the correct year, month and day. No need to sanitise and split/ run other regexes...

Reply:
yes, it works and removes additional padding